core/common/src/test/java/alluxio/UnderFileSystemFactoryRegistryRuleTest.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio;
+
+import org.junit.Test;
+
+/**
+ * Unit tests for {@link UnderFileSystemFactoryRegistryRule}.
+ */
+public class UnderFileSystemFactoryRegistryRuleTest {
+  //import part
+  //import alluxio.underfs.UnderFileSystemFactory;
+  //import alluxio.underfs.UnderFileSystemFactoryRegistry;
+  //import org.mockito.Mockito;
+  //import org.junit.runners.model.Statement;
+
+  //codepart
+  //private UnderFileSystemFactory mUnderFileSystemFactory;
+  //private UnderFileSystemFactoryRegistry mUnderFileSystemFactoryRegistry;
+  //private Statement mStatement = new Statement() {
+    //@Override
+    //public void evaluate() throws Throwable {
+      //mUnderFileSystemFactoryRegistry = Mockito.mock(UnderFileSystemFactoryRegistry.class);
+      //Mockito.when(mUnderFileSystemFactory.getClass().getMethod("register")).thenReturn();
+      //Mockito.when(mUnderFileSystemFactory.getClass().getMethod("unregister")).thenReturn();
+    //}
+  //};
+
+  @Test
+  public void testUnderFileSystemFactoryRegistryRule() throws Throwable {
+    //mUnderFileSystemFactory = Mockito.mock(UnderFileSystemFactory.class);
```

Comment:
Mockito is very powerful. Check here: http://www.baeldung.com/mockito-behavior

In our case, you can mock the factory to claim support any 
```java
Mockito.when(mUnderFileSystemFactory.supportsPath("mock://foo")).thenReturn(true);
```
Then in the `evaluate` part, you can claim the factory is registered
```java
public void evaluate() throws Throwable {
  assertEquals(mUnderFileSystemFactory , UnderFileSystemFactoryRegistry.find("mock://foo"));
}
```